djangocms_pageadmin/filters.py

```diff
@@ -71,8 +74,13 @@ class TemplateFilter(admin.SimpleListFilter):
     parameter_name = "template"
 
     def lookups(self, request, model_admin):
-        for value, name in get_cms_setting('TEMPLATES'):
-            yield (value, name)
+        site = get_current_site()
+        site_templates = getattr(settings, "SITE_TEMPLATES", {})
+        templates = site_templates.get(site.domain, None)
+        if templates:
```

Comment:
I looked into the issue that you were having earlier, and it was due to how python handles `return` and `yield` in the same function, which results in behaviour you might not initially expect. This helped me understand better what was going on: https://stackoverflow.com/questions/28806583/is-it-good-practice-to-mix-yield-and-return-statements-in-python

So separating into two methods like you have done gets around that. However it could be simplified further by removing the use of loops/yield entirely, and returning the settings directly:
```
        if templates:
            return templates
        else:
            return get_cms_setting('TEMPLATES')
```
This would then allow you to remove the helper methods.

Reply:
Done :-)